addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -388,23 +384,25 @@ public void pressButton(String device, String button) {
         }
     }
 
-    /**
-     * Cache a call to get the harmony config so we don't overwhelm it with requests.
-     *
-     * @return
-     */
-    public synchronized HarmonyConfig getCachedConfig() {
-        if (client == null) {
-            return null;
-        }
-
-        Date now = new Date();
-        if (cachedConfig == null || cacheConfigExpireDate == null || now.after(cacheConfigExpireDate)) {
-            logger.debug("Refreshing conf cache");
-            cachedConfig = client.getConfig();
-            cacheConfigExpireDate = new Date(System.currentTimeMillis() + CONFIG_CACHE_TIME);
-        }
-        return cachedConfig;
+    public synchronized CompletableFuture<HarmonyConfig> getConfigFuture() {
+        return configCache.getValue(() -> {
+            return new CompletableFuture<HarmonyConfig>() {
+                {
```

Comment:
Are these extra braces needed?

Reply:
It has been reworked as well. I've also retested that it still executes asynchronously with the right threads and properly throws exceptions.